powerline/segments/common/bat.py

```diff
@@ -65,13 +70,20 @@ def _get_battery(pl):
 			cap_path = linux_bat_fmt.format(linux_bat)
 			if linux_bat.startswith('BAT') and os.path.exists(cap_path):
 				pl.debug('Using /sys/class/power_supply with battery {0}', linux_bat)
-
-				def _get_capacity(pl):
-					with open(cap_path, 'r') as f:
-						return int(float(f.readline().split()[0]))
-
-				return _get_capacity
-		pl.debug('Not using /sys/class/power_supply as no batteries were found')
+				with open(cap_path, 'r') as f:
+					_capacity = int(float(f.readline().split()[0]))
+
+		linux_ac_fmt = '/sys/class/power_supply/{0}/online'
+		for linux_ac in os.listdir('/sys/class/power_supply'):
+			online_path = linux_ac_fmt.format(linux_ac)
+			if linux_ac.startswith('AC') and os.path.exists(online_path):
+				pl.debug('Using /sys/class/power_supply with ac {0}', linux_ac)
+				with open(online_path, 'r') as f:
+					_ac_powered = bool(f.readline())
```

Comment:
Also how the contents of the file looks like? If this is something like `0` or `1` then this code is incorrect: `bool("0") == bool("1") == (len("0") != 0) == (len("1") != 0) == True`.